src/providers/config/AppConfigurationProvider.tsx

```diff
@@ -0,0 +1,68 @@
+import isEmpty from 'lodash/isEmpty';
+import isNil from 'lodash/isNil';
+import React, { createContext, FunctionComponent, useContext, useMemo, useState } from 'react';
+
+export type VerdaccioOptions = {
+  url_prefix: string;
+  uri?: string;
+  base: string;
+  language?: string;
+  version?: string;
+  protocol?: string;
+  host?: string;
+  scope: string;
+  title: string;
+  logo?: string;
+  primaryColor: string;
+  darkMode: boolean;
+};
+
+type ConfigProviderProps = {
+  configOptions: VerdaccioOptions;
+  setConfigOptions: any;
+};
+
+const defaultValues: ConfigProviderProps = {
+  configOptions: {
+    primaryColor: '#4b5e40',
+    darkMode: false,
+    scope: '',
+    base: '',
+    url_prefix: '',
+    title: 'Verdaccio',
+  },
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  setConfigOptions: () => {},
+};
+
+function getConfiguration() {
+  const uiConfiguration = window?.__VERDACCIO_BASENAME_UI_OPTIONS ?? defaultValues.configOptions;
+  if (isNil(uiConfiguration.primaryColor) || isEmpty(uiConfiguration.primaryColor)) {
+    uiConfiguration.primaryColor = '#4b5e40';
+  }
+
+  return uiConfiguration;
+}
+
+const AppConfigurationContext = createContext<ConfigProviderProps>(defaultValues);
+
+const AppConfigurationProvider: FunctionComponent = ({ children }) => {
```

Comment:
I was wondering if we should keep using `FunctionComponent`.. just thinking :)

Reply:
Yes, otherwise children yells `Binding element 'children' implicitly has an 'any' type.ts(7031)`

Reply:
sure thing, but maybe we could write regular functions and add the type children to the Interfaces 🤷‍♀️ it's just an idea.